descartes_samplers/include/descartes_samplers/evaluators/impl/timing_edge_evaluator.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef DESCARTES_SAMPLERS_EVALUATORS_IMPL_TIMING_EDGE_EVALUATOR_HPP
+#define DESCARTES_SAMPLERS_EVALUATORS_IMPL_TIMING_EDGE_EVALUATOR_HPP
+
+#include <descartes_light/descartes_macros.h>
+DESCARTES_IGNORE_WARNINGS_PUSH
+#include <cmath>
+DESCARTES_IGNORE_WARNINGS_POP
+
+#include <descartes_samplers/evaluators/timing_edge_evaluator.h>
+
+namespace descartes_light
+{
+template <typename FloatType>
+TimingEdgeEvaluator<FloatType>::TimingEdgeEvaluator(const std::vector<FloatType>& velocity_limits,
+                                                    FloatType dt,
+                                                    FloatType safety_factor)
+  : EdgeEvaluator<FloatType>(velocity_limits.size())
+  , velocity_limits_(Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>>(
+        velocity_limits.data(),
+        static_cast<Eigen::Index>(velocity_limits.size())))
+  , dt_(dt)
+  , safety_factor_(safety_factor)
+{
+}
+
+template <typename FloatType>
+std::pair<bool, FloatType> TimingEdgeEvaluator<FloatType>::considerEdge(const FloatType* start, const FloatType* end)
+{
+  Eigen::Matrix<FloatType, Eigen::Dynamic, 1> delta =
+      Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>>(end, static_cast<Eigen::Index>(this->dof_)) -
+      Eigen::Map<const Eigen::Matrix<FloatType, Eigen::Dynamic, 1>>(start, static_cast<Eigen::Index>(this->dof_));
+  Eigen::Matrix<FloatType, Eigen::Dynamic, 1> joint_times = delta.cwiseQuotient(velocity_limits_).cwiseAbs();
```

Comment:
FYI you can use `Eigen::Array` instead of `Eigen::Matrix` to do coefficient-wise operations "natively" without calling `cwise...`

Reply:
I assumed they did the same thing under the hood. Is there a benefit for one over the other?

Reply:
`Eigen::Matrix` is for linear algebra operations whereas `Eigen::Array` is for coefficient-wise operations. According to the documentation, you can convert one to the other with no run-time cost. I think it's just more convenient to use one over the other depending on what type of operations you're trying to do